### DISCUSSION TITLE
Does Penpot use websockets and if so, which Apache directives are necessary for a reverse proxy?

### DISCUSSION THREAD

**Discussion by Verhoeckx** [uxbox/uxbox#2929] [Q&A] [answered]
Hello Penpot developers,

Question: does Penpot use websockets?

And if so, which Apache directives are needed in a reverse proxy?

For OpenVSCode I use the code below. Do I need to add something similar?

```
RewriteEngine On
RewriteCond %{HTTP:Connection} Upgrade [NC]
RewriteCond %{HTTP:Upgrade} websocket [NC]
RewriteRule /(.*) ws://localhost:2999/$1 [P,L]
```

This is, of course, only necessary if Penpot uses websockets.

**Comment by niwinz** [comment, marked as answer]
Yes, penpot uses websockets, but I'm personally not very familiar with apache, so I can't really help you on this area.

**Reply by niwinz** [reply]
Consider opening issues or ask on the community, this discussion are deprecated.

**Reply by Verhoeckx** [reply]
In hindsight, I will look for the official documentation of websockets and Apache and configure it myself.
Good to know that Penpot does use websockets!

Next time I will the use the Community forum! Maybe you can block Github Discussions for new posts?

**Reply by Verhoeckx** [reply]
For people who will find this thread, I used the code below to proxy the websockets.

```
    RewriteEngine On
    RewriteCond %{HTTP:Connection} Upgrade [NC]
    RewriteCond %{HTTP:Upgrade} websocket [NC]
    RewriteRule /(.*) ws://localhost:9001/$1 [P,L]
```
